packages/flutter/lib/src/cupertino/action_sheet.dart

```diff
@@ -0,0 +1,425 @@
+  /// [ActionSheetAction] widget in this entry can have [isCancelAction]
+  /// set to true.
+  final List<Widget> actions;
+
+  /// A scroll controller that can be used to control the scrolling of the
+  /// [message] in the action sheet.
+  ///
+  /// This attribute is typically not needed, as alert messages should be
+  /// short.
+  final ScrollController messageScrollController;
+
+  /// A scroll controller that can be used to control the scrolling of the
+  /// [actions] in the action sheet.
+  ///
+  /// This attribute is typically not needed.
+  final ScrollController actionScrollController;
+
+  ///NEEDS DOC
+  final VoidCallback cancelCallback;
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> content = <Widget>[];
+
+    if (title != null || message != null) {
+      final Widget titleSection = new _CupertinoActionSheetTitleSection(
+        title: title,
+        message: message,
+        messageScrollController: messageScrollController,
+      );
+      content.add(new Flexible(child: titleSection));
+    }
+
+    content.add(new Flexible(child: new _CupertinoActionSheetActionSection(
+      children: actions,
+      actionScrollController: actionScrollController,
+    )));
+
+    final List<Widget> columnContents = <Widget>[];
+
+    columnContents.add(
+      new ClipRRect(
+        borderRadius: const BorderRadius.all(const Radius.circular(12.0)),
+        child: new DecoratedBox(
+          decoration: _kCupertinoDialogBackFill,
+          child: new BackdropFilter(
+            filter: new ImageFilter.blur(sigmaX: 10.0, sigmaY: 10.0),
+            child: new Container(
+              decoration: _kCupertinoDialogFrontFillDecoration,
+              child: new Column(
+                mainAxisSize: MainAxisSize.min,
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+                children: content,
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    if (cancelCallback != null) {
+      columnContents.add(const Padding(
+        padding: const EdgeInsets.only(top: 8.0),
+      ));
+      columnContents.add(
+        new ClipRRect(
+          borderRadius: const BorderRadius.all(const Radius.circular(12.0)),
+          child: new DecoratedBox(
+            decoration: _kCupertinoDialogBackFill,
+            child: new BackdropFilter(
+              filter: new ImageFilter.blur(sigmaX: 10.0, sigmaY: 10.0),
+              child: new Container(
+                height: _kButtonHeight,
+                decoration: _kCupertinoDialogFrontFillDecoration,
+                child: new ActionSheetAction(
+                  child: const Text('Cancel'),
+                  onPressed: cancelCallback,
+                  isDefaultAction: true,
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 8.0),
+      child: new Center(
```

Comment:
Here's your problem with getting the child size in your layout's getPositionForChild.  The Center is expanding to fill the whole available area. Just remove this Center.

Reply:
Perfect, that worked! Do you think making that change will help with my sizing problem if I continue to play around with constraints?

Reply:
Yes, I think it should.  Give it a try.